src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1872,29 +1901,8 @@ public void execute() throws GitException, InterruptedException {
      * @throws hudson.plugins.git.GitException if underlying git operation fails
      * @throws java.lang.InterruptedException if interrupted
      */
-    public Set<Branch> getRemoteBranches() throws GitException, InterruptedException {
-        try (Repository db = getRepository()) {
-            Map<String, Ref> refs = db.getAllRefs();
-            Set<Branch> branches = new HashSet<>();
-
-            for(Ref candidate : refs.values()) {
-                if(candidate.getName().startsWith(Constants.R_REMOTES)) {
-                    Branch buildBranch = new Branch(candidate);
-                    if (!GitClient.quietRemoteBranches) {
-                        listener.getLogger().println("Seen branch in repository " + buildBranch.getName());
-                    }
-                    branches.add(buildBranch);
-                }
-            }
-
-            if (branches.size() == 1) {
-                listener.getLogger().println("Seen 1 remote branch");
-            } else {
-                listener.getLogger().println(MessageFormat.format("Seen {0} remote branches", branches.size()));
-            }
-
-            return branches;
-        }
+    public Set<Branch> getRemoteBranches() throws GitException, InterruptedException, IOException {
+        return parseBranches(launchCommand("branch", "-r", "-v", "--no-abbrev"));
```

Comment:
wonder why the initial code was relying on JGit within the CLIGit implementation. Would prefer not to mix such changes in this PR, but had no choice here.